web/client/src/components/infrastructure/infrastructure-map.vue

```diff
@@ -0,0 +1,287 @@
+<template>
+    <div ref="container">
+        <div
+            ref="map"
+            class="map infrastructure-map"
+        />
+        <div
+            ref="mapLegend"
+            class="map-overlay infrastructure-map-legend"
+        >
+            <template
+                v-for="(elementType, index) in legendControlTypes"
+                :key="index"
+            >
+                <input
+                    :id="elementType"
+                    :ref="elementType"
+                    :value="elementType"
+                    :checked="initiallyCheckedControls.includes(elementType)"
+                    type="checkbox"
+                    @input="onLegendControlClicked"
+                >
+                <label
+                    class="legend-key"
+                    :for="elementType"
+                >
+                    <img
+                        v-if="hasImage(elementType)"
+                        class="legend-key-icon"
+                        :src="iconUrl + elementType + iconExtension"
+                        alt=""
+                    >
+                    {{ elementTypeLabels[elementType] ?? elementType }}
+                </label>
+                <br>
+            </template>
+        </div>
+
+        <div
+            ref="infrastructureTooltip"
+            class="infrastructureTooltip infrastructure-tooltip"
+        >
+            <ul id="infrastructureTooltipList">
+                <li id="kilometerPoint" />
+                <li id="risingOrFalling" />
+            </ul>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { mapState } from 'vuex';
+import { InfrastructureNamespace } from '@/stores/infrastructure-store';
+import {
+    deHighlightSignalStationRoute,
+    deHighlightStationRoute,
+    highlightSignalStationRoute,
+    highlightStationRoute
+} from './infrastructureMap';
+import { FilterSpecification, Map } from 'maplibre-gl';
+import { infrastructureMapStyle } from './mapStyle';
+import { addIcons, iconExtension, iconUrl } from './addIcons';
+import { elementTypes, elementTypeLabels } from './elementTypes';
+import { defineComponent } from 'vue';
+
+const specialLayoutControls = ['Rising', 'Falling'];
+const initiallyCheckedControls = ['station', 'ms', 'as', 'eotd', ...specialLayoutControls];
+const legendControlTypes = [
+    ...elementTypes,
+    ...specialLayoutControls
+];
+
+const mapDefaults = {
+    style: infrastructureMapStyle,
+    attributionControl: false,
+    zoom: 18,
+    hash: 'location',
+    center: [8, 47],
+    maxBounds: [[6, 45], [17, 55]], // [SW Point] [NE Point] in LonLat
+    bearing: 0,
+};
+
+export default defineComponent({
+    name: 'InfrastructureMap',
+
+    data() {
+        return {
+            libreGLMap: null as (Map | null),
```

Comment:
I think it should be 
```suggestion
            libreGLMap: unknown as (Map | null),
```

Reply:
?

![image](https://user-images.githubusercontent.com/78490564/211167782-7b8100b9-b500-426a-8e31-2b8a8564523b.png)